src/main/java/oof/command/FreeCommand.java

```diff
@@ -133,41 +146,145 @@ private void parseOutput(Ui ui) throws OofException {
     }
 
     /**
-     * Populates the lists for tracking time and events.
+     * Parses the output for finding free time slots if there are occupied slots.
+     *
+     * @param ui Prints relevant output.
+     * @throws OofException Prints customised error message.
+     */
+    private void parseOutput(Ui ui) throws OofException {
+        try {
+            for (int i = 0; i < TOTAL_TIME_SLOTS; i++) {
+                if (slotStates.get(i) == SLOT_FREE) {
+                    ui.printFreeSlots(startingTimeSlots[i], endingTimeSlots[i]);
+                } else if (slotStates.get(i) == SLOT_BUSY) {
+                    ui.printBusySlots(startingTimeSlots[i], endingTimeSlots[i]);
+                }
+            }
+            if (isSuggestionBlockPresent() && !sortedDeadlineNames.isEmpty()) {
+                ui.printSuggestionDetails(sortedDeadlineNames);
+            }
+        } catch (DateTimeException e) {
+            throw new OofException("Timestamp given is invalid! Please try again.");
+        }
+    }
+
+    /**
+     * Checks the list of slot states for a consecutive 4 hour free block.
+     * @return true if there is a 4 hour free block, false otherwise
+     */
+    private boolean isSuggestionBlockPresent() {
+        int countConsecutive = 0;
+        for (int i = 0; i < TOTAL_TIME_SLOTS; i++) {
+            if (SLOT_FREE == slotStates.get(i)) {
+                countConsecutive++;
+            } else {
+                countConsecutive = 0;
+            }
+            if (countConsecutive == SUGGESTIONS_BLOCK) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Populates the lists for tracking event times.
      *
-     * @param dateStart     Start date of task.
+     * @param dateStart     Start date of event.
+     * @param dateEnd       End date of event.
      * @param freeSlotsDate Date inputted by user.
-     * @param startTime     Starting time of task.
-     * @param endTime       Ending time of task.
-     * @param event         Event task object.
-     * @throws OofException Prints customised exception message.
+     * @param startTime     Starting time of event.
+     * @param endTime       Ending time of event.
+     * @throws OofException   Prints customised exception message.
      */
-    private void populateList(String dateStart, String freeSlotsDate, String startTime, String endTime, Event event)
-            throws OofException {
+    private void populateEventTimes(String dateStart, String dateEnd, String freeSlotsDate, String startTime,
+                                    String endTime) throws OofException {
```

Comment:
Violates coding standard used in CS2113T, change the method signature wrapping to 8 spaces.